src/components/studies/app-design/AppDesign.tsx

```diff
@@ -564,19 +511,20 @@ const AppDesign: React.FunctionComponent<
                   checked={
                     !appDesignProperties.welcomeScreenInfo.isUsingDefaultMessage
                   }
-                  onChange={() =>
-                    setAppDesignProperties(prevState => {
-                      const newWelcomeScreenData = {
-                        ...prevState.welcomeScreenInfo,
-                      }
-                      newWelcomeScreenData.isUsingDefaultMessage =
-                        !prevState.welcomeScreenInfo.isUsingDefaultMessage
-                      return {
-                        ...appDesignProperties,
-                        welcomeScreenInfo: newWelcomeScreenData,
-                      }
+                  onChange={() => {
```

Comment:
on Change should be called with the actual params of the things that you are changing rather than updating the appDesignProperties reference. I think you should get away from using the appDesignProperties. Just updated the straight up values